src/plugins/data/server/search/search_service.ts

```diff
@@ -327,7 +331,7 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
   };
 
   private deleteSession = async (deps: SearchStrategyDependencies, sessionId: string) => {
-    this.cancelSessionSearches(deps, sessionId);
+    await this.cancelSessionSearches(deps, sessionId);
```

Comment:
wonder why we await here, but not in `cancelSession`. should we also await there?